### ISSUE TITLE
Ratio and offset on all sensors

### ISSUE THREAD

**Issue by WhiskeyGolf** [FrSkyRC/ETHOS-Feedback-Community#4885]
It would be nice if you could set the offset and ratio in all telemetry sensors (except RSSI and VFR), as is already possible with voltage. 
I need this function to adjust conical servo temperature values ​​and much more. 
Many thanks

**Comment by bsongis-frsky** [comment]
Agreed that we need an offset for temperatures

**Comment by WhiskeyGolf** [comment]
It would be nice if it were at least possible to set an offset/ratio in the current sensors. I have deviations of 20% in consumption.

**Comment by CpeDratDoPC62** [comment]
I agree and have the same problem: I cannot compensate the telemetry sensor values to the correct values, as it was possible in OpenTX!

**Comment by CpeDratDoPC62** [comment]
So, I managed to calibrate the current measurement for FAS 7A in Ethos 1.6.2 and also speed up the voltage reading.

I managed to speed up the really slow voltage responses to the actual value by replaying FW 1.5, which is designed for FAS 7A ADV, and I had FAS 7A Lite. Now it has started to measure voltage as quickly as the other sensors.

Now to compensate for the current range:
I needed to reduce the value by about 2.5%, which is missing for current, and this is also missing in ETHOS 1.6.3 everywhere except for the voltage sensor values.
I got around this by setting one calculated sensor. Add the calculated value to the current sensor, but you cannot use the directly entered coefficient for multiplication or division (another error in Ethos), you have to use Vars.
Use the raw value from the Current sensor as input.
Simply multiplying by 98.5% set in Var does not give the correct result, and it also divides by ten (another error in Ethos), so when compensating for a current that had a real value of 5.000A, the raw value measured on the Current sensor was 5.1A, which I reduced by the coefficient, but be careful, by dividing by the calculated sensor value Var1 102.5%, it was reduced to 0.500A, but somehow the division by ten unfortunately comes into play. I then removed this by adding a second division by another Var value, set to 10.0%, and the calculated sensor began to show the correct and very accurate value of 5.0A at a current in the range of 4.96 - 5.05A. 
I then found that 0.000A was not at the correct value, so I used the zero button (its use is not described anywhere by Frsky). After disconnecting the circuit so that no current flows, the button is held down for 30 seconds, and as a result, the resulting value of 0.1A suddenly began to display correctly in the actual current range from 0.05-0.14A, and if we raise it to 0.16A, the value on the calculated sensor flashes between 0.1-0.2, and so I am satisfied with how precisely it works! " :-)
It's just a pain that this cannot be set directly in Ethos, as is possible in Open TX. Tears in my eyes.... :-(

**Comment by WhiskeyGolf** [comment]
I've had these problems with the calculated sensors a lot, too. I sometimes wonder if none of the developers actually use them. Otherwise, it would be obvious how many errors there are in the Cala sensors and how complicated these workarounds are. The developers actually aim to keep Ethos very simple and straightforward. But if I'm supposed to do the whole process with CalcSensors for 10 Xact servos instead of an offset, I find Ethos much more complicated than, say, OTX.

**Comment by CpeDratDoPC62** [comment]
So today, while combing through the telemetry, I found out what is causing the error in the calculated sensor!
If I set the source to two decimal places, in my case it is Current from FAS 7A, so that it shows a value of 1.20A (here, for example, 1.000A flows through the sensor and it incorrectly shows 1.200A), and I want the calculated sensor to show 1. 00A, I have to divide this value by Var 120% and the result after this division will be 0. 10A, but if I cancel the two decimal places in the source sensor from FAS 7A to one, so that it no longer shows an unnecessary zero at the end, then it is 1.2A and then the division in the calculated sensor by the number Var 120% calculates the correct result, which is 1.0A!!!!
It is very sensitive to the number of decimal places in the input value – if a different number of decimal places is used than is set as the default for the measured value, this can lead to incorrect mathematical results. It seems that whoever wrote the program probably did not test this with a different number of decimal places than is set as standard.

**Comment by Voglberg** [comment]
Hello [CpeDratDoPC62](https://github.com/CpeDratDoPC62) , can you please set the VAR here? I would like to recreate it with a Sequre ESC. This has a deviation of 10% in the consumption sensor. Thank you
